tests/integration/olp-cpp-sdk-dataservice-read/StreamLayerClientTest.cpp

```diff
@@ -0,0 +1,455 @@
+                                            olp::http::HttpStatusCode::OK),
+                                        HTTP_RESPONSE_LOOKUP_STREAM));
+
+  ON_CALL(*network_mock_,
+          Send(IsPostRequest(URL_STREAM_SUBSCRIBE_SERIAL), _, _, _, _))
+      .WillByDefault(
+          ReturnHttpResponse(olp::http::NetworkResponse().WithStatus(
+                                 olp::http::HttpStatusCode::CREATED),
+                             HTTP_RESPONSE_STREAM_LAYER_SUBSCRIPTION));
+
+  ON_CALL(*network_mock_,
+          Send(IsPostRequest(URL_STREAM_SUBSCRIBE_PARALLEL), _, _, _, _))
+      .WillByDefault(
+          ReturnHttpResponse(olp::http::NetworkResponse().WithStatus(
+                                 olp::http::HttpStatusCode::CREATED),
+                             HTTP_RESPONSE_STREAM_LAYER_SUBSCRIPTION));
+
+  ON_CALL(*network_mock_,
+          Send(IsPostRequest(URL_STREAM_SUBSCRIBE_SUBSCRIPTION_ID), _, _, _, _))
+      .WillByDefault(
+          ReturnHttpResponse(olp::http::NetworkResponse().WithStatus(
+                                 olp::http::HttpStatusCode::CREATED),
+                             HTTP_RESPONSE_STREAM_LAYER_SUBSCRIPTION));
+
+  ON_CALL(*network_mock_,
+          Send(IsPostRequest(URL_STREAM_SUBSCRIBE_CONSUMER_ID), _, _, _, _))
+      .WillByDefault(
+          ReturnHttpResponse(olp::http::NetworkResponse().WithStatus(
+                                 olp::http::HttpStatusCode::CREATED),
+                             HTTP_RESPONSE_STREAM_LAYER_SUBSCRIPTION));
+
+  ON_CALL(*network_mock_,
+          Send(IsPostRequest(URL_STREAM_SUBSCRIBE_ALL_PARAMETERS), _, _, _, _))
+      .WillByDefault(
+          ReturnHttpResponse(olp::http::NetworkResponse().WithStatus(
+                                 olp::http::HttpStatusCode::CREATED),
+                             HTTP_RESPONSE_STREAM_LAYER_SUBSCRIPTION));
+
+  // Catch any non-interesting network calls that don't need to be verified
+  EXPECT_CALL(*network_mock_, Send(_, _, _, _, _)).Times(AtLeast(0));
+}
+
+TEST_F(ReadStreamLayerClientTest, Subscribe) {
+  HRN hrn(GetTestCatalog());
+
+  {
+    SCOPED_TRACE("Subscribe succeeds, serial");
+
+    EXPECT_CALL(*network_mock_,
+                Send(IsGetRequest(URL_LOOKUP_STREAM), _, _, _, _))
+        .Times(1);
+
+    EXPECT_CALL(*network_mock_,
+                Send(IsPostRequest(URL_STREAM_SUBSCRIBE_SERIAL), _, _, _, _))
+        .Times(1);
+
+    StreamLayerClient client(hrn, kLayerId, settings_);
+
+    SubscribeResponse subscribe_response;
+    Condition condition;
+    client.Subscribe(SubscribeRequest(), [&](SubscribeResponse response) {
+      subscribe_response = std::move(response);
+      condition.Notify();
+    });
+
+    ASSERT_TRUE(condition.Wait(kTimeout));
+
+    ASSERT_TRUE(subscribe_response.IsSuccessful())
+        << ApiErrorToString(subscribe_response.GetError());
+
+    EXPECT_EQ(kSubscriptionId, subscribe_response.GetResult().c_str());
+  }
```

Comment:
I guess you should clear and verify the expectations of network after each `SCOPED_TRACE`

Reply:
Added